conjureup/charm.py

```diff
@@ -20,40 +19,45 @@ def get_file(bundle, dst):
     req = requests.get(bundle)
     if not req.ok:
         raise Exception("Could not query file in charmstore: {}".format(req))
-    return req.json()
+    return req.text
 
 
-def get_bundle(bundle, to_file=False):
+def get_bundle(bundle_name, channel='stable'):
     """ Attempts to grab the bundle.yaml
 
     Arguments:
-    bundle: URL of bundle or absolute path to local bundle
+    bundle_name: name of bundle (ie canonical-kubernetes)
+    channel: Which channel to pull the bundle from
     to_file: store to a temporary file
 
     Returns:
     Dictionary of bundle's yaml unless to_file is True,
     then returns the path to the downloaded bundle
     """
-    if path.isfile(bundle):
-        if to_file:
-            with NamedTemporaryFile(mode="w", encoding="utf-8",
-                                    delete=False) as tempf:
-                shutil.copyfile(bundle, tempf.name)
-            return tempf.name
-        else:
-            with open(bundle) as f:
-                return yaml.safe_load(f.read())
+    bundle_channel_info = get_channel_info(bundle_name, channel)
+    bundle = "{}-{}".format(bundle_name,
+                            bundle_channel_info['Revision'])
+    return yaml.safe_load(get_file(bundle, 'bundle.yaml'))
 
-    req = requests.get(bundle)
+
+def get_channel_info(bundle_name, channel='stable'):
+    """ Grabs the channel information for a bundle
+
+    Usage:
+    https://api.jujucharms.com/charmstore/v5/canonical-kubernetes/meta/id?channel=edge
+
+    Arguments:
+    bundle_name: name of bundle (ie canonical-kubernetes)
+    channel: the release channel (ie stable, candidate, beta, edge)
+    """
+    query = path.join(cs, bundle_name,
+                      "meta/id?channel={}".format(channel))
+    req = requests.get(query)
```

Comment:
When we looked at it before I think it required like libsodium or some type of binding for it to work.